File: app/src/main/java/bzh/naoned/naonedbusdatadisplay/service/model/TrafficData.java
package bzh.naoned.naonedbusdatadisplay.service.model;

import com.google.gson.annotations.SerializedName;

import java.text.DateFormat;
import java.util.Date;
import java.util.Locale;

import io.realm.RealmList;
import io.realm.RealmObject;

public class TrafficData extends RealmObject {
    @SerializedName("id")
    private String mId;
    @SerializedName("title")
    private String mTitle;
    @SerializedName("message")
    private String mMessage;
    @SerializedName("dateStart")
    private Date mDate;
    @SerializedName("sourceCode")
    private String mSourceCode;
    @SerializedName("sourceLabel")
    private String mSourceLabel;
    @SerializedName("sourcePicture")
    private String mSourcePicture;
    @SerializedName("targets")
    private RealmList<Target> mTargets;

    public String getId() {
        return mId;
    }

    public String getTitle() {
        return mTitle;
    }

    public String getMessage() {
        return mMessage;
    }

    public String getDate() {
        return new java.text.SimpleDateFormat("dd/MM/yyyy à HH:mm", Locale.getDefault()).format(mDate);
    }

    public String getSourceCode() {
        return mSourceCode;
    }

    public String getSourceLabel() {
        return mSourceLabel;
    }

    public String getSourcePicture() {
        return mSourcePicture;
    }

    public RealmList<Target> getTargets() {
        return mTargets;
    }
}

File: app/src/main/java/bzh/naoned/naonedbusdatadisplay/service/model/TrafficData.kt
package bzh.naoned.naonedbusdatadisplay.service.model

import com.google.gson.annotations.SerializedName

import java.text.DateFormat
import java.util.Date
import java.util.Locale

import io.realm.RealmList
import io.realm.RealmObject

open class TrafficData : RealmObject() {
    @SerializedName("id")
    var id: String? = null
    @SerializedName("title")
    var title: String? = null
    @SerializedName("message")
    var message: String? = null
    @SerializedName("dateStart")
    private var mDate: Date? = null
    @SerializedName("sourceCode")
    var sourceCode: String? = null
    @SerializedName("sourceLabel")
    var sourceLabel: String? = null
    @SerializedName("sourcePicture")
    var sourcePicture: String? = null
    @SerializedName("targets")
    var targets: RealmList<Target>? = null

    var date: String? = null
        get() = java.text.SimpleDateFormat("dd/MM/yyyy à HH:mm", Locale.getDefault()).format(mDate)
}
